test/unittest/participant/participant_factory/ParticipantFactoryTest.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest_aux.hpp>
+#include <gtest/gtest.h>
+
+#include <ddsrouter/communication/PayloadPool.hpp>
+#include <ddsrouter/configuration/ParticipantConfiguration.hpp>
+#include <ddsrouter/configuration/SimpleParticipantConfiguration.hpp>
+#include <ddsrouter/dynamic/DiscoveryDatabase.hpp>
+#include <ddsrouter/exceptions/ConfigurationException.hpp>
+#include <ddsrouter/participant/IParticipant.hpp>
+#include <ddsrouter/participant/ParticipantFactory.hpp>
+#include <ddsrouter/types/participant/ParticipantId.hpp>
+#include <ddsrouter/types/participant/ParticipantType.hpp>
+
+using namespace eprosima::ddsrouter;
+using namespace eprosima::ddsrouter::rtps;
+
+namespace eprosima {
+namespace ddsrouter {
+namespace test {
+
+std::shared_ptr<IParticipant> create_participant(
+        std::string id_str)
+{
+    ParticipantFactory participant_factory;
+    std::shared_ptr<PayloadPool> payload_pool = std::make_shared<PayloadPool>();
+    std::shared_ptr<DiscoveryDatabase> discovery_database = std::make_shared<DiscoveryDatabase>();
+    ParticipantId id(id_str);
+    ParticipantConfiguration* participant_configuration;
+    if (id_str == "local")
+    {
+        ParticipantConfiguration parent_participant_configuration(id);
+        participant_configuration = new SimpleParticipantConfiguration(parent_participant_configuration);
+    }
+    else
+    {
+        participant_configuration = new ParticipantConfiguration(id);
+    }
```

Comment:
This is not needed. Every Participant create its own Configuration from `ParticipantConfiguration` (this is why copy constructor from parent is required in every configuration, another OOP failure :cow2: ).
```suggestion
    participant_configuration = new ParticipantConfiguration(id);
```

 What it may be required is to add specific fields for some Configurations (i.e. domain in Simple) otherwise the Participant  creation should fail. (with PR #64 it will be required).